dev/NavigationView/NavigationView.cpp

```diff
@@ -4704,6 +4739,71 @@ bool NavigationView::IsFullScreenOrTabletMode()
     return isFullScreenMode || isTabletMode;
 }
 
+void NavigationView::SetDropShadow()
+{
+    const auto displayMode = DisplayMode();
+
+    if (displayMode == winrt::NavigationViewDisplayMode::Compact || displayMode == winrt::NavigationViewDisplayMode::Minimal)
+    {
+        if (const auto shadowCaster = m_shadowCaster.get())
+        {
+            const auto rootSplitView = m_rootSplitView.get();
+            const auto rootSplitViewActualWidth = rootSplitView.ActualWidth();
+
+            // If the screen real estate is too small, the left pane of SplitView ends up being smaller than OpenPaneLength and takes the width of RootSplitView.
+            // This requires a different storyboard animation than default.
+            if (rootSplitViewActualWidth < rootSplitView.OpenPaneLength())
+            {
+                // Since the width of the NavigationView pane is now different than OpenPaneLength, we need to update its TranslateX animation value to a reflect that.
+                const auto negativeSplitViewWidthMinusCompactLength = (rootSplitViewActualWidth - rootSplitView.CompactPaneLength()) * -1;
+                GetTemplateSettings()->NegativeSplitViewWidthMinusCompactLength(negativeSplitViewWidthMinusCompactLength);
+
+                if (const auto shadowCasterSmallPaneEaseInStoryboard = m_shadowCasterSmallPaneEaseInStoryboard.get())
+                {
+                    shadowCasterSmallPaneEaseInStoryboard.Begin();
+                }
+            }
+            else
+            {
+                if (const auto shadowCasterEaseInStoryboard = m_shadowCasterEaseInStoryboard.get())
+                {
+                    shadowCasterEaseInStoryboard.Begin();
+                }
+            }
+
+            if (winrt::IUIElement10 shadowCaster_uiElement10 = shadowCaster)
+            {
+                shadowCaster_uiElement10.Shadow(winrt::ThemeShadow{});
+            }
+        }              
+    }
+}
+
+void NavigationView::UnsetDropShadow()
+{
+    const auto shadowCaster = m_shadowCaster.get();
+
+    if (const auto shadowCasterEaseOutStoryboard = m_shadowCasterEaseOutStoryboard.get())
+    {
+        shadowCasterEaseOutStoryboard.Begin();
+        shadowCasterEaseOutStoryboard.Completed(winrt::auto_revoke,
```

Comment:
It looks like you aren't saving the revoker returned by this method anywhere. This will cause the revoker to be cleaned up as soon as the program leaves this scope, which will cause the event to get unregistered. Looking at this code I don't think it works, did you test that the ShadowCasterEaseOutStoryboard_Completed gets called?

Reply:
Oops you're right. I stored it in a member variable.